client/src/modules/journal/modals/search.modal.js

```diff
@@ -25,36 +49,58 @@ function JournalSearchModalController(Instance, Users, Projects, Notify, options
     })
     .catch(Notify.handleError);
 
-  vm.onSelectAccount = function onSelectAccount(account) {
-    vm.options.account_id = account.id;
-  };
-
   // format voucher types and bind to the view
   Vouchers.transactionType()
     .then(function (list) {
-      // make sure that the items are translated
-      list.data.forEach(function (item) {
+      vm.types = list.data.map(function (item) {
         item.hrText = $translate.instant(item.text);
+        return item;
       });
-
-      // bind to the view
-      vm.types = list.data;
     })
     .catch(Notify.handleError);
 
-  // deletes a filter from the options/parameters
-  vm.clear = function clear(key) {
-    delete vm.options[key];
+  // handle component selection states
+  // custom filter account_id - assign the value to the searchQueries object
+  vm.onSelectAccount = function onSelectAccount(account) {
+    vm.searchQueries.account_id = account.id;
   };
 
-  vm.cancel = function cancel() {
-    Instance.dismiss();
+  // deafult filter period - directly write to changes list
+  vm.onSelectPeriod = function onSelectPeriod(period) {
+    var periodFilters = Periods.processFilterChanges(period);
+
+    periodFilters.forEach(function (filterChange) {
+      changes.post(filterChange);
+    });
+  };
+
+  // default filter limit - directly write to changes list
+  vm.onSelectLimit = function onSelectLimit(value) {
+    // input is type value, this will only be defined for a valid number
+    if (angular.isDefined(value)) {
+      changes.post({ key : 'limit', value : value });
+    }
   };
 
+  // deletes a filter from the custom filter object, this key will no longer be written to changes on exit
+  vm.clear = function clear(key) {
+    delete vm.searchQueries[key];
+  };
+
+  vm.cancel = Instance.dismiss;
+
   // returns the filters to the journal to be used to refresh the page
   vm.submit = function submit(form) {
-    if (form.$invalid) { return; }
+    // push all searchQuery values into the changes array to be applied
+    angular.forEach(vm.searchQueries, function (value, key) {
```

Comment:
This is certainly possible! The way that the changes array currently works expects all values to be used even if they haven't been changed ($dirty) - we would just have to make sure they're all picked up. This could be refined as this is implemented across modules.